src/datreant/core/limbs.py

```diff
@@ -87,17 +88,25 @@ def _setter(self, val):
             raise TypeError("Can only set with tags, a list, or set")
 
     def __getitem__(self, value):
+        # check if we might have a string to parse into a selection object
+        if isinstance(value, string_types):
+            value = Parser.parseString(value)[0].generate_tag_expr()
+        return self._getselection(value)
+
+    def _getselection(self, value):
+        """get tags according to a selection
+        """
         with self._read:
             if isinstance(value, list):
                 # a list of tags gives only members with ALL the tags
-                fits = all([self[item] for item in value])
+                fits = all([self._getselection(item) for item in value])
```

Comment:
With the `any` and `all` calls, you can pass in a generator (ie remove the square brackets) so they can exit early